src/Arcus.WebApi.Security/Authentication/CertificateAuthenticationAttribute.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Security.Cryptography.X509Certificates;
+using GuardNet;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Arcus.WebApi.Security.Authentication
+{
+    /// <summary>
+    /// Authentication filter to secure HTTP requests by allowing only certain values in the client certificate.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = true)]
+    public class CertificateAuthenticationAttribute : TypeFilterAttribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CertificateAuthenticationAttribute"/> class.
+        /// </summary>
+        /// <param name="requirement">The property of the client <see cref="X509Certificate2"/> to validate.</param>
+        /// <param name="expectedValue">The expected value the property of the <see cref="X509Certificate2"/> should have.</param>
+        public CertificateAuthenticationAttribute(X509ValidationRequirement requirement, string expectedValue) : base(typeof(CertificateAuthenticationFilter))
```

Comment:
`CertificateAuthenticationAttribute` should not depend on `CertificateAuthenticationFilter`, I see this more as both of these using a shared certificate authentication class as part of it.

Reply:
Not sure what you mean with this. Structuring this way is the recommended way to do this.
It is only dependend on the type to create instances, not on a single (thus shared) instance.

Reply:
I think @tomkerkhove means 'favor composition over inheritance' ?

Reply:
That is correct, sorry for not being clear!

Reply:
Hmm, still not clear to me. As I said, this way of defining both local and global access policies are default approaches because of the exact same implementation.

Reply:
That is correct, but one approach depends on the other where my suggestion is to move out the effective validation logic into a seperate class which is used by both local and global access policies.

Let's say a `ICertificateValidator` which is used by both `CertificateAuthenticationAttribute` and `CertificateAuthenticationFilter`

Reply:
Ok, you could move the validation logic; but that doesn't change the fact that the attribute still requires a filter implementation to work properly; and will therefore still requires the certificate filter.
That's just how type filters work; so the only real change is yet another abtraction layer that is used only in a single class.

Reply:
Validation logic was moved to dedicated class, resolving